### ISSUE TITLE
NumberBox AutomationProperties.Name and Min/Max is Ignored by AutomationPeer

### ISSUE THREAD

**Issue by robloo** [microsoft/microsoft-ui-xaml#2951]
**Describe the bug**

Setting a NumberBox's AutomationProperties.Name attached property is ignored by its AutomationPeer. The Minimum/Maximum values are also not used but should be.

**Steps to reproduce the bug**

1. Set the AutomationProperties.Name attached property of a NumberBox in XAML
2. Using the Windows Narrator, focus the NumberBox (and therefore the InputTextBox inside it)
3. The NumberBox AutomationPeer and Narrator will not use the AutomationProperties.Name when the InputTextBox is focused
4. The NumberBox AutomationPeer and Narrator will not use any Minimum/Maximum values

**Expected behavior**

This should behave the same as the Slider. The AutomationProperties.Name should be used first then a description of the value. Additionally, the AutomationPeer should really give the min/max values of the range as well. The Slider just gives a percent but the NumberBox needs to be more specific.

**Screenshots**
N/A

**Version Info**

NuGet package version: 
Microsoft.UI.Xaml 2.4.0

**Comment by StephenLPeters** [comment]
@robloo we had this issue https://github.com/microsoft/microsoft-ui-xaml/issues/1854 which says it was fixed.  Seems related but not exactly the same?

**Comment by robloo** [comment]
@StephenLPeters Yea, thought it sounded familiar. I should have looked in more detail. That said, I'm expanding it to also add the Minimum/Maximum values as well as that's pretty important for accessibility.

I ran across this over the weekend with WinUI 2.4.0. I'll update to 2.4.2 and test again for the Name property.

**Comment by StephenLPeters** [comment]
I looked at the change and I do think it should fix the name issue. @YuliKl is there someone from the accessibility team we can ask about what to do about the min and max values?

**Comment by ranjeshj** [comment]
The API looks similar to RangeBase - should the peer support IRangeValueProvider ?

**Comment by robloo** [comment]
@StephenLPeters I tested this again with latest 2.4.2 of WinUI -- it still has an issue and doesn't read aloud the AutomationProperties.Name.

"Edit zero, zero selected" is all I get with the below code.

```xaml
<mux:NumberBox
  x:Name="Channel1NumberBox"
  AutomationProperties.Name="Red Channel"
  Grid.Column="1"
  Grid.Row="1"
  CornerRadius="0,2,2,0"
  VerticalAlignment="Center" />
```

I also notices that IRangeValueProvider is implemented already 
https://github.com/microsoft/microsoft-ui-xaml/blob/86a7d62bf8141a1f3e141d3ade7a69bce02063c1/dev/NumberBox/NumberBoxAutomationPeer.cpp#L66

I think the issue might be related to focus. It's behaving as if only the `InputBox` is focused -- and it is but only as a part of the NumberBox. NumberBoxAutomationPeer is not used and instead narrator is just using the default AutomationPeer for TextBox.

**Comment by YuliKl** [comment]
@SavoySchuler FYI. Perhaps you can reach out to someone on the Accessibility team for help.
I'm also wondering whether #1882 is related.

**Comment by MikeHillberg** [comment]
>  I think the issue might be related to focus. It's behaving as if only the InputBox is focused -- and it is but only as a part of the NumberBox. NumberBoxAutomationPeer is not used and instead narrator is just using the default AutomationPeer for TextBox (which doesn't have an AutomationProperties.Name set as that is part of NumberBox).

I think that's exactly right. You'd expect to have the same issue with AutoSuggestBox, but in its OnApplyTemplate is forwards the Automation name to the TextBox part. It looks like editable ComboBox needs the same treatment.

More generally maybe we should teach TextBoxAutomationPeer to work with its templated parent when it's acting as a template part. In the example of NumberBox, it needs to be a range provider for the sake of min/max.

**Comment by chingucoding** [comment]
@MikeHillberg Is there anything we can do in WinUI 2 to prevent the TextBoxAutomationPeer to interfere with us here and allow us to pass Min/Max to UIA?

@ranjeshj Would it be fine if I create a PR to pass the AutomationProperties.Name to the TextBox so we can fill that gap? Also, when AutomationProperties.Name is not set, but the header of the NumberBox is set, should we pass the header value as AutomationProperties.Name value? After all, that is one of the ways the TextBox gets its UIA name.

**Comment by ranjeshj** [comment]
@chingucoding lets give the same treatment for automation name in the mean time (and open a separate issue to track TextBoxAutomationPeer to understand the templated parent scenario).

**Comment by chingucoding** [comment]
@ranjesh Just to be sure, so should we apply to the same header handling logic to numberbox or should we just pass the `AutomationProperties.Name` value to the textbox?

**Comment by ranjeshj** [comment]
Do we have the header handling logic in WinUI already ? If so, lets match that pattern. I was thinking we can lookup the automationproperties.name and use if if it is valid, if not fallback to the header.

**Comment by chingucoding** [comment]
I think we have a helper for this that get's us the appropriate UIA name depending on the properties set. Though this would require us to listen to changes on the `AutomationProperties.Name` property so we can update the textboxes UIA name accordingly.
